vars/waitForHelixRuns.groovy

```diff
@@ -23,168 +23,180 @@ def call (def helixRunsBlob, String prStatusPrefix) {
     def mcUrlMap = [:]
     def failedRunMap = [:]
     def passed = true
+    timestamps {
+        // Wrap in the default timeout of 120 mins
+        timeout(15) {
+            waitUntil(minRecurrencePeriod: 60, maxRecurrencePeriod: 180, unit: 'SECONDS') {
```

Comment:
Why do we need two levels of waitUntil here?  If we end up outside the for loop that means that either
1) We got all the results.
2) We hit the timeout (in which case we'd skip the second level).

My take here is that we should have a relatively short waitUntil (5-10 seconds between) inside the for loop with a simple timeout wrapping the whole thing.

Alternatively, you could add another parallel block to do this work.